internal/cli/service_pull_image.go

```diff
@@ -0,0 +1,56 @@
+package cli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	cliTypes "github.com/docker/cli/cli/config/types"
+)
+
+func (s Service) PullImage(config PullImageConfig) error {
```

Comment:
Lots of overlap here (and in the tests) with the pulling and tagging parts of `rwx image build` but I'm thinking we keep them distinct for now while some of these product interactions are still being fleshed out and can always extract later.